src/NuGet.Clients/PackageManagement.UI/Utility/TelemetryUtility.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Common;
+using NuGet.ProjectManagement;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Some utility apis for telemetry operations.
+    /// </summary>
+    public static class TelemetryUtility
+    {
+        /// <summary>
+        /// Create ActionTelemetryEvent instance.
+        /// </summary>
+        /// <param name="projects"></param>
+        /// <param name="operationType"></param>
+        /// <param name="source"></param>
+        /// <param name="startTime"></param>
+        /// <param name="status"></param>
+        /// <param name="statusMessage"></param>
+        /// <param name="packageCount"></param>
+        /// <param name="endTime"></param>
+        /// <param name="duration"></param>
+        /// <returns></returns>
+        public static ActionsTelemetryEvent GetActionTelemetryEvent(
```

Comment:
It could but then the purpose of this method will change besides i'll also need to pass telemetry service instance to emit event. Here i just wanted to construct the event object to avoid duplicate code.